### PR TITLE
Footer

### LINKED ISSUE #2: Footer
The footer appears on every page and is managed via a fragment.  It has the following features:
- Fixed disclaimer, authored as default content in the footer.
- Links, organized in sections.  Each of these sections should be a row in a footer nav block.  The list of links should be managed as a set of links for that section.  For now we don't have the pages imported so the links can be to # but for testing formatting/appearance we should author the links
- Social media links (if this is too complex, can break into a new story)
- Language switcher, should use same logic as header language switcher.  Note that the current language for the site is not listed.  Use a placeholder XLSX file to hold the i18n labels.
- Bottom links for terms of use, legal notics, privacy policy and copyright.  This should be authored as default content.

NOTE: This story does not include the HELP or MyMomentiveBot features, ignore those for now.

image

Mobile:
- In mobile/responsive view, the four columns turn into summary blocks which only expand one at a time

image

Note that the current website has an issue formatting the list of languages correctly, you have some room to explore a better language selector here (or use the one from header) -- The current experience needs improvement.

### PR DESCRIPTION
**Pull request by samuelopra**
Fix #2 

Test URLs:
- Before: https://main--momentive--aemsites.hlx.live
- After: https://issue-2--momentive--aemsites.hlx.live

### CODE REVIEW

File: blocks/footer/footer.css [hunk 1]
```diff
@@ -1,14 +1,91 @@
...
-  padding: 2rem;
-  background-color: var(--light-color);
-  font-size: var(--body-font-size-s);
-}
-
 footer .footer {
-  max-width: 1200px;
+  max-width: 100vw;
   margin: auto;
+  background-color: #1a355e;
```
Comment (badvision) [inline blocks/footer/footer.css:4]: See styles.css - this should be `var(--navy-blue)`

File: blocks/footer/footer.css [hunk 2]
```diff
@@ -1,14 +1,91 @@
...
+  color: #fff;
```
Comment (badvision) [inline blocks/footer/footer.css:5]: color: white or color: var(--background-color) would be preferred here.  It seems pedantic but realisically, if they change their entire color scheme it makes it a lot more convenient to have everything based on variables.

File: blocks/footer/footer.css [hunk 3]
```diff
@@ -1,14 +1,91 @@
...
+.footer-sublinks-row h3 {
+ margin-top: 0;
+}
+
+.footer-sublinks-row .default-content-wrapper {
+  color: #1a355e;
+}
+
+.footer-sublinks-row a {
+  color: #1a355e;
```
Comment (badvision) [inline blocks/footer/footer.css:61]: Same comment as above, use the navy-blue variable

File: blocks/footer/footer.css [hunk 4]
```diff
@@ -1,14 +1,91 @@
...
+  display: inline-block;
+  padding: 0 20px;
+}
+
+.footer-main h3 {
+  font-size: 17px;
```
Comment (badvision) [inline blocks/footer/footer.css:67]: Better to use var(--heading-font-size-xs) or var(--body-font-size-xs) if possible.  This also allows better uniformity across breakpoints as we often adjust fonts up or down according to screen size.

File: blocks/footer/footer.css [hunk 5]
```diff
@@ -1,14 +1,91 @@
...
+}
+
+.footer ul {
+  padding: 0;
   margin: 0;
+  list-style: none;
+}
+
+.footer li {
+  font-size: 17px;
```
Comment (badvision) [inline blocks/footer/footer.css:80]: Same comment as above for font sizes

File: blocks/footer/footer.css [hunk 6]
```diff
@@ -1,14 +1,91 @@
...
+  padding-left: 0;
+  padding-bottom: 10px;
+}
+
+.footer-main a {
+  color: #fff;
+}
+
+.footer-social-wrapper a {
+  color: #fff;
```
Comment (badvision) [inline blocks/footer/footer.css:90]: Same comment as above, use var:(--background-color)

File: blocks/footer/footer.js [hunk 7]
```diff
@@ -9,13 +9,62 @@ export default async function decorate(block) {
...
 
+  // Directly select the .footer-main container to append the columns to
+  const footerMain = footer.querySelector('.footer-main');
+  const section = footer.querySelector('.section');
+
+  // Find each <h3> within .footer-main
+  const mainSections = footerMain.querySelectorAll('h3');
+  mainSections.forEach((h3) => {
+    // Identify the next <ul> before any changes
+    const nextUl = h3.nextElementSibling && h3.nextElementSibling.tagName === 'UL'
```
Comment (badvision) [inline blocks/footer/footer.js:26]: Can simplify to h3?.nextElementSibling.tagName === 'UL'

**Comment by badvision** [comment]
So far coming along pretty well, just make sure we're using the variables declared in style.css for uniformity.  Also, important to keep in mind the default styles should be for the smallest (mobile) layout, and then anything needed for larger screen sizes should be declared with media selector (e.g. @media (width>=900px), etc) -- there's a lot of examples of this in the styles.css and other blocks already if you need to look at those for ideas.  left-rail does a lot of funky stuff to deal with different behaviors in mobile vs desktop, but hopefully we don't have to do that kind of thing too much.

File: blocks/footer/footer.css [hunk 8]
```diff
@@ -1,14 +1,258 @@
...
+.footer-sublinks-row li {
+  font-family: var(--avenir-demi);
+  font-size: var(--body-font-size-xs);
+  color: var(--navy-blue);
+  padding: 0 20px;
 }
 
-footer .footer p {
+.footer-social-row {
+  max-width: 100vw !important;
```
Comment (badvision) [inline blocks/footer/footer.css:105]: Generally you shouldn't have to use !important -- that's usually a sign that something else is taking priority that shouldn't, or that your selectors are not specific enough so they're not getting priority.

File: blocks/footer/footer.js [hunk 9]
```diff
@@ -9,13 +9,67 @@ export default async function decorate(block) {
...
+    const nextUl = h3.nextElementSibling?.tagName === 'UL' ? h3.nextElementSibling : null;
+
+    const wrapperDiv = document.createElement('div');
+    wrapperDiv.classList.add('footer-column');
+
+    wrapperDiv.appendChild(h3);
+    if (nextUl) {
+      wrapperDiv.appendChild(nextUl);
+      h3.classList.add('toggle');
+      nextUl.classList.add('hidden'); // Initially hide the `ul` elements
```
Comment (badvision) [inline blocks/footer/footer.js:34]: This is fine with me, but one thing to consider is to make things hidden by default in the CSS, and then visible only when a class is applied.  That way, they don't show up before your code executes.  This avoids CLS issues.

**Review by badvision — APPROVED** [review]
Approved, but consider trying to improve by removing !important where possible.

File: blocks/footer/footer.js (hunk 9 above)
Comment (samuelopra) [reply blocks/footer/footer.js]: I understand your point here. I refactored to make this hidden by default

**Pull request merged**